src/main/org/firebirdsql/gds/ng/StatementOperationAware.java

```diff
@@ -0,0 +1,51 @@
+package org.firebirdsql.gds.ng;
+
+import org.firebirdsql.gds.Operation;
+
+/**
+ * Abstract class for controlling the execution of statement
+ *
+ * @author <a href="mailto:vasiliy.yashkov@red-soft.ru">Vasiliy Yashkov</a>
+ */
+public abstract class StatementOperationAware {
+    // Implementation of this class that controls the execution of operations
+    private static StatementOperationAware instance;
+
+    /**
+     * If there is an instance of class, then inform about operation starting
+     * @param op operation
+     */
+    public static void startStatementOperation(Operation op) {
+        if (instance != null)
```

Comment:
Instead of the `null`-check, I suggest to use a 'null-object' with an implementation that doesn't do anything.